src/OpenTelemetry/Trace/Tracer.cs

```diff
@@ -69,9 +69,9 @@ internal Tracer(SpanProcessor spanProcessor, Sampler sampler, TracerConfiguratio
 
         public IDisposable WithSpan(ISpan span, bool endSpanOnDispose)
         {
-            if (span == null)
-            {
-                throw new ArgumentNullException(nameof(span));
+            if (span != null)
+            { 
```

Comment:
whitespace

Reply:
? what is the problem here?

Reply:
oh, I see. I'm ok to fix it, but I'd prefer it either be enforced by the tooling or not enforced at all.

Reply:
#372 includes a setting in .editorconfig to remove trailing whitespace, which will automatically be applied once merged.